### ISSUE TITLE
Symlink to supermarket-ctl missing when upgrading supermarket 2.4.2 RPM to 2.5.0

### ISSUE THREAD

**Issue by zimmma00** [chef/supermarket#1265]
I down loaded the new supermarket code (2.5).  When I upgrade it (from 2.4.2), it removed the symbolic link /bin/supermarket-ctl and did not put it back.  I had to do that manually.  The 2.4.2 built this symbolic link.  

I am running rhel 7.2.

When installing supermarket 2.5 the symbolic link is created, but when upgrading from 2.4.2, the symbolic link is not created.

Is this a packaging issue?

**Comment by robbkidd** [comment]
I'm not sure. We did not see this problem with several upgrades of Ubuntu hosts. I'll investigate.

**Comment by zimmma00** [comment]
This would have been with .rpm rather than with the .deb.

**Comment by robbkidd** [comment]
Yep. That's one of the aspects I'll be looking at.

**Comment by robbkidd** [comment]
For informational purposes, the replication steps:

### install 2.4.2 - links are there
```
[vagrant@default-centos-72 ~]$ sudo rpm -i /tmp/packages/supermarket-2.4.2-1.el7.x86_64.rpm
warning: /tmp/packages/supermarket-2.4.2-1.el7.x86_64.rpm: Header V4 DSA/SHA1 Signature, key ID 83ef826a: NOKEY

Thank you for installing Supermarket!

The next step in the process is to run:

supermarket-ctl reconfigure

[vagrant@default-centos-72 ~]$ ls -l /bin/supermarket-ctl
lrwxrwxrwx. 1 root root 36 Apr 11 17:39 /bin/supermarket-ctl -> /opt/supermarket/bin/supermarket-ctl
[vagrant@default-centos-72 ~]$ ls -l /usr/bin/supermarket-ctl
lrwxrwxrwx. 1 root root 36 Apr 11 17:39 /usr/bin/supermarket-ctl -> /opt/supermarket/bin/supermarket-ctl
```

### upgrade to 2.5.0 - links are not present
```
[vagrant@default-centos-72 ~]$ sudo rpm -U --replacepkgs /tmp/packages/supermarket-2.5.0-1.el7.x86_64.rpm
warning: /tmp/packages/supermarket-2.5.0-1.el7.x86_64.rpm: Header V4 DSA/SHA1 Signature, key ID 83ef826a: NOKEY

Thank you for installing Supermarket!

The next step in the process is to run:

supermarket-ctl reconfigure

[vagrant@default-centos-72 ~]$ ls -l /bin/supermarket-ctl
ls: cannot access /bin/supermarket-ctl: No such file or directory
[vagrant@default-centos-72 ~]$ ls -l /usr/bin/supermarket-ctl
ls: cannot access /usr/bin/supermarket-ctl: No such file or directory
```

### upgrade to 2.5.0 again with the same command - links are back
```
[vagrant@default-centos-72 ~]$ sudo rpm -U --replacepkgs /tmp/packages/supermarket-2.5.0-1.el7.x86_64.rpm
warning: /tmp/packages/supermarket-2.5.0-1.el7.x86_64.rpm: Header V4 DSA/SHA1 Signature, key ID 83ef826a: NOKEY

Thank you for installing Supermarket!

The next step in the process is to run:

supermarket-ctl reconfigure

[vagrant@default-centos-72 ~]$ ls -l /usr/bin/supermarket-ctl
lrwxrwxrwx. 1 root root 36 Apr 11 18:02 /usr/bin/supermarket-ctl -> /opt/supermarket/bin/supermarket-ctl
[vagrant@default-centos-72 ~]$
```

**Comment by robbkidd** [comment]
Same missing symlink behavior between 2.4.1 and 2.4.2.

**Comment by robbkidd** [comment]
```
$ inotifywait -e create,delete,modify,move -mrq /usr/bin &

$ sudo rpm -U /tmp/packages/supermarket-2.5.0-1.el7.x86_64.rpm
warning: /tmp/packages/supermarket-2.5.0-1.el7.x86_64.rpm: Header V4 DSA/SHA1 Signature, key ID 83ef826a: NOKEY
/usr/bin/ DELETE supermarket-ctl
/usr/bin/ CREATE supermarket-ctl

Thank you for installing Supermarket!

The next step in the process is to run:

supermarket-ctl reconfigure

/usr/bin/ DELETE supermarket-ctl
```

**Comment by robbkidd** [comment]
I believe I've figured it out. During an upgrade, RPM runs the old packages post-uninstall script as the last step. We learned this for the Chef and Chef Server omnibus packages, but that didn't carry over to Supermarket. PR coming soon.

@zimmma00 In the meantime, there is a work-around: run the `rpm -U --replacepkgs supermarket-2.5.0-1.el7.x86_64.rpm` command a second time, the symlinks will be present.
